atmat/atphysics/Radiation/check_6d.m

```diff
@@ -1,42 +1,55 @@
-function [is_6d,ring]=check_6d(ring,enable,varargin)
-%CHECK_6D	Check the presence of longitudinal motion in a lattice
+function [is_6d, ring] = check_6d(ring,enable,varargin)
+%CHECK_6D Checks the presence of longitudinal motion in a lattice.
 %
-%IS_6D=CHECK_6D(RING) Return the radiation state of RING (true/false).
-%   Equivalent to IS_6D=<a href="matlab:help atGetRingProperties">atGetRingProperties</a>(RING,'is_6d')
+%IS_6D = CHECK_6D(RING) Returns the radiation state of RING (true/false).
+%  Equivalent to IS_6D=<a href="matlab:help atGetRingProperties">atGetRingProperties</a>(RING,'is_6d')
 %
-%IS_6D=CHECK_6D(RING,ENABLE) Generate an error if IS_6D is different of ENABLE
+%IS_6D = CHECK_6D(RING, ENABLE)
+%  Generates an error if IS_6D is different of ENABLE
 %
-% ENABLE:    Desired 6D state
+%[IS_6D, NEWRING] = CHECK_6D(RING,ENABLE,'force')
+%  The keyword 'force' overrides any error check, and converts
+%  the RING according to the value of ENABLE.
+%  IS_6D contains the status of the RING before conversion.
 %
-%[IS_6D,NEWRING]=CHECK_6D(RING,ENABLE,'force')
-%   Convert RING to the desired state using default options
-%
-% ENABLE:    Desired 6D state
+%[IS_6D, NEWRING] = CHECK_6D(RING,ENABLE,'strict',STRICT)
+%  Default, STRICT=true.
+%  If STRICT is true, a difference btw IS_6D and ENABLE produces an error.
+%  If STRICT is false, a difference btw IS_6D and ENABLE produces a warning,
+%  and the RING is converted according to the value of ENABLE.
+%  IS_6D contains the status of the RING before conversion.
 %
 % See also: ATGETRINGPROPERTIES, ATENABLE_6D, ATDISABLE_6D
 
-force=getflag(varargin,'force');
-is_6d=atGetRingProperties(ring,'is_6d');
+[force, varargs] = getflag(varargin, 'force');
+[strict, ~] = getoption(varargs, 'strict', true);
+is_6d = atGetRingProperties(ring, 'is_6d');
 
-if nargin > 1
-    if xor(is_6d,enable)
-        if force
-            if enable
-                ring=atenable_6d(ring);
-            else
-                ring=atdisable_6d(ring);
-            end
-        else
-            error('AT:Radiation',['''is_6d'' must be ' boolstring(enable)])
-        end
-    end
+if nargin == 1, return; end
+
+if ~xor(is_6d,enable), return; end
+
+if ~force
+    outerrmsg = join([" is_6d must be" boolstring(enable) newline]);
```

Comment:
`error` also needs the `'AT: radiation'` as a 1st argument, before the message, Otherwise the Id is empty. This error Id is useful in try...catch constructs to identify different errors.